Software/bsw/static/Mcal/CAN/src/Can.c

```diff
@@ -1230,88 +1225,104 @@ Std_ReturnType Can_SetControllerMode(uint8 Controller,
 	return ret;
 }
 
-Std_ReturnType Can_GetControllerMode(uint8 Controller,
-		Can_ControllerStateType* ControllerModePtr) {
+Std_ReturnType Can_GetControllerMode(uint8 Controller,Can_ControllerStateType* ControllerModePtr) 
+{
 	Std_ReturnType Loc_Can_GetControllerMode_Ret = E_OK;
 
 #if(CAN_DEV_ERROR_DETECT == STD_ON)
-	if (Can_ControllerState[Controller]==CAN_CS_UNINIT)
+	if (ControllerState[Controller]==CAN_CS_UNINIT)
 	{
 		Det_ReportError(CAN_MODULE_ID,CAN_INSTANCE_ID,Can_GetControllerMode_Id,CAN_E_UNINIT);
 		Loc_Can_GetControllerMode_Ret = E_NOT_OK;
 	}
 	else
 	{
-
+	    /* Do Nothing */
 	}
 
 #endif
 
 	if (Controller >= NUM_OF_CAN_CONTROLLERS)
 	{
-		Det_ReportError(CAN_MODULE_ID, CAN_INSTANCE_ID,
-				Can_GetControllerMode_Id, CAN_E_PARAM_CONTROLLER);
+		Det_ReportError(CAN_MODULE_ID, CAN_INSTANCE_ID,Can_GetControllerMode_Id, CAN_E_PARAM_CONTROLLER);
 		Loc_Can_GetControllerMode_Ret = E_NOT_OK;
-	} else {
-
+	} 
+	else 
+	{
+	    /* Do Nothing */
 	}
 	if (ControllerModePtr == NULL)
 	{
-		Det_ReportError(CAN_MODULE_ID, CAN_INSTANCE_ID,
-				Can_GetControllerMode_Id, CAN_E_PARAM_POINTER);
+		Det_ReportError(CAN_MODULE_ID, CAN_INSTANCE_ID,Can_GetControllerMode_Id, CAN_E_PARAM_POINTER);
 		Loc_Can_GetControllerMode_Ret = E_NOT_OK;
-	} else {
-
+	} 
+	else
+	{
+	    /* Do Nothing */
 	}
 
 	if (Loc_Can_GetControllerMode_Ret != E_NOT_OK)
 	{
 
-		*ControllerModePtr = Can_ControllerState[Controller];
-	} else {
-
+		*ControllerModePtr = ControllerState[Controller];
 	}
+	else
+	{
+	    /* Do Nothing */
+	}
+
 	return Loc_Can_GetControllerMode_Ret;
 
 }
 
-void Can_MainFunction_Write(void) {
+void Can_MainFunction_Write(void) 
+{
 
 #if ((CAN_TX_PROCESSING_0==POLLING_PROCESSING) || (CAN_TX_PROCESSING_1==POLLING_PROCESSING)\
```

Comment:
I think performing one global checking on the controller ID then checking the TX processing type rather than multiple if conditions that perform a previously done work, would save execution time and make the code more readable and portable. e.g. 
`set global variable that holds the current controller ID.
 gather the duplicate code in a single private function.
 check the controller ID:
       if ID = 0: then
          call private function to check processing type()
          call private function to confirm successful TX()
      else if ID == 1
          ...`

Reply:
Controller activation and processing type Checking in the pre- processing phase, so I think this will not affect the execution time